### ISSUE TITLE
Trigger multiple times when press trim

### ISSUE THREAD

**Issue by frazebee** [jrwieland/F3A#7]
Hi,

I find out there's some problems when this widget running on my radio.
I set up all the logical swtiches and special funtions as your document decribe and checked they all worked normally.
But when I press the trim button, the sequence always goes forward/backward by 2 or 3 (from No.1 to No.3 or No.4). Did you have the same problem or not?
It seems the refresh function runs too frequently when the LS triggers.
Hope for your reply, thanks!

**Comment by frazebee** [comment]
https://github.com/user-attachments/assets/63b766fb-60d1-46db-8a14-1a7e205db50d

You may see from the video. No matter which position I put the switch, every event triggers 2 times.

**Comment by jrwieland** [comment]
I'll look at it this afternoon and see if I put a delay on the switch if it shows down the jump.
Thank you for letting me know about the issue

**Comment by frazebee** [comment]
> I'll look at it this afternoon and see if I put a delay on the switch if it shows down the jump. Thank you for letting me know about the issue

Thanks! I was also considering add some delay or pause in the refresh function, but I did not make it...

**Comment by jrwieland** [comment]
@frazebee I see your using an actual Horus.  Each radio type requires a little tweak.  I run TaraniTunes on RM16s as my main widget, it uses the similar controls (Rudder & aileron trims) , I have added a duration to my LS to control the "jumping" that occasionally occurred.  

The duration is dependent on how long you press your trims, for me .05 is the sweet spot. A smaller duration can occasionally still cause a jump and longer seems not to move for me unless I hold it.

On my older 9xd (B&W) the duration is not needed since it is running as a script and not a widget.
I don't believe I can control the refresh interval of a widget. 

I know it is not a perfect solution, but it should make it more usable than having it jump over the next call.
LMK how it goes for you.

**Comment by frazebee** [comment]
Actually I mainly use RM16s. Horus and RM show the same problem. I put the widget on Horus to see if there will be any difference but it seems not.
 I tried to modify the duration of the LS, but no help to this issue.  The shortest duration is 0.1 sec. The longer the make it even worse (jump more).

**Comment by jrwieland** [comment]
Are you in the US? If so it is possible to discuss it on the phone
John Wieland
412-789-5000

**Comment by frazebee** [comment]
Sorry, I'm in China and it's working hours now. Maybe we can discuss later
on Telegram or something else?
My telegram: +86 18221999294
Junzhe Ying

John Wieland ***@***.***> 于2024年8月12日周一 08:57写道：

> Are you in the US? If so it is possible to discuss it on the phone
> John Wieland
> 412-789-5000
>

**Comment by jrwieland** [comment]
I discovered the issue
I will have it operational tomorrow
Sorry for the delay
Funny thing it works in companion perfectly.
John Wieland
412-789-5000

**Comment by frazebee** [comment]
Thanks!

**Comment by frazebee** [comment]
I tried this on the companion, it works well....very interesting

**Comment by jrwieland** [comment]
Sorry for the delay I had that work thing get in the road.
Here is the repaired file
[main.txt](https://github.com/user-attachments/files/16592939/main.txt)
Rename the ending lua instead of txt
I will update the repository later and close the issue.
Until then I'll leave it open

I set the trigger for the repeat the maneuver to SC- it is switch #8, Switches run numerically from Sa+=1,SA-=2 etc.  If you are using a different switch as a source change the number in the file.
![repeat PNG](https://github.com/user-attachments/assets/6863f141-f09a-4de7-8355-30176309f2cc)


I added another globalValue to control the playlists (GV8) Named [GV7 in the code] if not needed or desired comment them out along with the triggers.
You will need to add additional SF's to change between the maneuver lists.

![screenshot_tx16s_24-08-12_20-41-35](https://github.com/user-attachments/assets/0e006f8a-97d9-482a-ac25-3aa128dc967d)

For testing I also added F25 and P25 to the CallList, I removed them and left only the AMA Advanced as a Call List, add others as needed.

This was fully tested on my actual RM16s 
LMK if you have any other issues
Again, thanks for letting me know about this one
